components/server-side-render/index.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	Component,
+	RawHTML,
+} from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { addQueryArgs } from '@wordpress/url';
+
+export class ServerSideRender extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			response: {},
+			attributes: props,
+		};
+	}
+
+	componentDidMount() {
+		this.getOutput();
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( JSON.stringify( nextProps ) !== JSON.stringify( this.props ) ) {
+			this.setState( { attributes: nextProps }, this.getOutput );
+		}
+	}
+
+	getOutput() {
+		this.setState( { response: {} } );
+		const { block } = this.props;
+		const attributes = this.state.attributes;
+		const apiURL = addQueryArgs( wpApiSettings.root + 'gutenberg/v1/blocks-renderer/' + block, {
+			...attributes,
+			_wpnonce: wpApiSettings.nonce,
+		} );
+		return window.fetch( apiURL, {
```

Comment:
May want to use `wp.apiRequest` for consistency. See #5253.

Reply:
I think more of a guide on using would help with that. I was working on another PR that used `wp.apiRequest` and it was leading to test failures (maybe it was how #4710 used in redux `connect`) so I removed, tests started passing and I could move forward. The only reason my modifications caused issues using `wp.apiRequest` as far as I could tell was that the tests themselves were not being passed enough context.